### DISCUSSION TITLE
WPF Report Viewer

### DISCUSSION THREAD

**Discussion by mmatriccino** [gerardo-lijs/CrystalReportsRunner#14] [General]
Have you ever tried to integrate the WPF report viewer?  I see that the assemblies for the WPF viewer are referenced but never used.  I am assuming that the Shell program would have to be rewritten.

**Comment by gerardo-lijs** [comment]
Hi, I don't understand your question about the WPF report viewer integration. Can you please clarify? Regarding the referenced assemblies, they are being used. Can you please clarify this second statement. And the third statement about the Shell program that has to be rewritten, I'm also lost here.

**Comment by mmatriccino** [comment]
Instead of using CrystalDecisions.Windows.Forms.CrystalReportViewer in a Windows Forms Dialog you would have a WPF Window that uses SAPBusinessObjects.WPF.Viewer.CrystalReportsViewer.   They are completely different viewers.

**Comment by gerardo-lijs** [comment]
Thanks for clarifying! It is possible to write your own "Runner" and use a WPF Report Viewer. There are some instructions here https://github.com/gerardo-lijs/CrystalReportsRunner/blob/main/docs/custom-runner.md.  This could use whatever version of Crystal Reports Runtime you want and target either 32 or 64bits.

Can I ask, why would you want or need the report viewer to use WPF? The library currently works fine with .NET 8 WPF app and the runner in Winforms. If there is some interesting feature/use/benefit, please let me know and I can help create this WPF Runner. It shouldn't be that hard really. If you want to write it yourself, PRs are also welcome!

**Reply by Tragen** [reply]
One of the reasons is that WPF can use OpenType fonts while WinForms can only use TrueType fonts
